### ISSUE TITLE
Searching for an email fails if the first_name attribute was removed

### ISSUE THREAD

**Issue by presidento** [pennersr/django-allauth#3001]
TL;DR: I think a simple check should be added to `get_user_search_fields` function here that the attribute is a field (or at least is not None):
https://github.com/pennersr/django-allauth/blob/ced1ddc730c36eca3551406c60e1577e30e01cbd/allauth/account/adapter.py#L425
It should be written: `lambda a: a and hasattr(user, a) and getattr(user, a) is not None`.

Rationale: when the project uses `AbstractUser` as a base class, it is recommended to set `first_name` to null if the project does not need it. So the user model will have the `first_name` attribute, but it will cause an error. In these cases I think it would be better to skip these common, but actually not used attributes.

**Comment by derek-adair** [comment]
Looks worthy to me, never hesitate to open random PR's in projects for simple improvements like this.  I'd imagine this will get merged rather quickly.

**Comment by pennersr** [comment]
Closing -- this is already fixed, `hasattr` is not used anymore.

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3001
